File: app/src/main/java/com/scatl/uestcbbs/module/dayquestion/adapter/DayQuestionAdapter.java
package com.scatl.uestcbbs.module.dayquestion.adapter;

import android.widget.RadioButton;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.DayQuestionBean;

/**
 * author: sca_tl
 * date: 2020/5/21 15:37
 * description:
 */
public class DayQuestionAdapter extends BaseQuickAdapter<DayQuestionBean.Options, BaseViewHolder> {

    private int checkedPosition;

    public DayQuestionAdapter(int layoutResId) {
        super(layoutResId);
    }

    public void setCheckedPosition(int position) {
        this.checkedPosition = position;
        notifyItemRangeChanged(0, getData().size());
    }

    public int getCheckedPosition() {
        return checkedPosition;
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, DayQuestionBean.Options item) {
        helper.addOnClickListener(R.id.item_day_question_radio_btn);
        RadioButton radioButton = helper.getView(R.id.item_day_question_radio_btn);
        radioButton.setText(item.dsp);
        radioButton.setChecked(helper.getLayoutPosition() == checkedPosition);

    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/dayquestion/adapter/DayQuestionAdapter.kt
package com.scatl.uestcbbs.module.dayquestion.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemDayQuestionBinding
import com.scatl.uestcbbs.entity.DayQuestionBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder

/**
 * Created by sca_tl at 2023/6/9 16:17
 */
class DayQuestionAdapter: PreloadAdapter<DayQuestionBean.Options, ItemDayQuestionBinding>() {

    var checkedPosition = 0
        set(value) {
            field = value
            notifyItemRangeChanged(0, items.size)
        }

    override fun getViewBinding(parent: ViewGroup): ItemDayQuestionBinding {
        return ItemDayQuestionBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemDayQuestionBinding>, position: Int, item: DayQuestionBean.Options?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.radioBtn.text = item.dsp
        holder.binding.radioBtn.isChecked = holder.layoutPosition == checkedPosition
    }
}